gsy_framework/schema/validators.py

```diff
@@ -0,0 +1,47 @@
+import abc
+from pathlib import Path
+
+import avro.schema
+from avro.errors import AvroTypeException
+from avro.io import validate as validate_avro_schema
+
+AVRO_SCHEMAS_PATH = Path(__file__).parent / "avro_schemas"
+
+
+class SchemaError(Exception):
+    """Exception class to be used by all defined schema validators."""
+    def __init__(self, message: str):
+        super().__init__(self)
+        self.message = message
+
+    def __str__(self):
+        return self.message
+
+
+class BaseSchemaValidator(abc.ABC):
+    """Base schema validator class to control inter-service communications."""
+
+    @abc.abstractmethod
+    def validate(self, data, raise_exception: bool = False) -> (bool, str):
+        """Validate given data using the defined schema."""
+
+
+class AVROSchemaValidator(BaseSchemaValidator):
+    """Schema validator class that utilizes """
+    def __init__(self, schema_name: str):
+        with open(AVRO_SCHEMAS_PATH / f"{schema_name}.json") as schema_file:
+            self.schema = avro.schema.parse(schema_file.read())
+
+    def validate(self, data, raise_exception: bool = False) -> (bool, str):
+        try:
+            validate_avro_schema(self.schema, data, raise_on_error=True)
+            return True, ""
+        except AvroTypeException as exc:
+            if raise_exception:
+                raise SchemaError(message="The provided data is invalid for the schema.") from exc
+            return False, str(exc)
+
+
+def get_schema_validator(schema_name: str) -> BaseSchemaValidator:
```

Comment:
I am not sure what do think about this interface WRT of usability. As a user of this function, I need to know which schema file I have to use.
IMO,[ this initiation](https://github.com/gridsingularity/gsy-e/pull/1598/files#diff-334c6d546ca90b8e453ec70f72cff8ead015cd681337d29179dab26338858fcbR48-R49) should be done here in the gsy_framework. This can be re-used on the gsy-web side, too.

Reply:
I admit that the initiation can be done in gsy-framework and be used in other services, but then we have to initialize every schema (with all their different versions) here. And this is something I'm trying to avoid as there may be different versions of each schema in the future and obviously initiating all of them in all of the services will use extra amounts of memory. It's important to point out that each service may be tied to an specific version of the schema in the future independently - and this is something to be decided by the service and not gsy-framework.

Reply:
OK, you convinced me here. Leave it as is. Thanks!